bridge-token-factory/src/lib.rs

```diff
@@ -98,12 +102,67 @@ pub trait FungibleToken {
 #[ext_contract(ext_bridge_token)]
 pub trait ExtBridgeToken {
     fn mint(&self, account_id: AccountId, amount: U128);
+
+    fn ft_transfer_call(
+        &mut self,
+        receiver_id: ValidAccountId,
+        amount: U128,
+        memo: Option<String>,
+        msg: String,
+    ) -> PromiseOrValue<U128>;
 }
 
 pub fn assert_self() {
     assert_eq!(env::predecessor_account_id(), env::current_account_id());
 }
 
+enum Recipient {
+    AccountId(AccountId),
+    ContractId {
+        evm_address: AccountId,
+        message: String,
+    },
+}
+
+/// `recipient` is the target account id receiving current ERC-20 tokens.
+///
+/// If `recipient` doesn't contain a semicolon (:) then it is interpreted as a NEAR account id
+/// and token are minted as NEP-141 directly on `recipient` account id.
+///
+/// Otherwise, the format expected is: <target_address>:<message>
+///
+/// @target_address: Account id of the contract to transfer current funds
+/// @message: Free form message to be send to the target using ft_transfer_call
+///
+/// The final message sent to the `target_address` has the format:
+///
+/// <message>:<predecessor_account_id>
+///
+/// Where `message` is the free form string that was passed,
+/// then a semicolon (:) as a separator between `message` and `predecessor_account_id`
+/// and `predecessor_account_id` is the caller of this function.
+fn parse_recipient(recipient: String) -> Recipient {
+    if recipient.contains(':') {
+        let mut iter = recipient.split(':');
+        let evm_address = iter.next().unwrap().into();
+        let message = format!(
+            "{}:{}",
+            iter.collect::<Vec<&str>>().join(":"),
+            env::predecessor_account_id()
+        );
+        Recipient::ContractId {
+            evm_address,
+            message,
+        }
+    } else {
+        Recipient::AccountId(recipient)
+    }
```

Comment:
```suggestion
    if let Some(index) = recipient.find(':') {
        Recipient {
            recipient[.. index].into(),
            Some(format!("{}:{}", recipient[index + 1 ..], env::predecessor_account_id())),
        }
    } else {
        Recipient { recipient, None }
    }
```
Also, what's the point of passing `predecessor_account_id`?
1. Because this function is called from `finish_deposit`, `predecessor_account_id` will always be the token factory account ID.
2. Even if this were called from within `deposit`, it will be the account ID of whoever relayed the message. So, this can only be used for reimbursing gas fees, but this functionality probably belongs to the connector itself, so no need to pass this to the recipient.